eventlet/greenpool.py

```diff
@@ -1,18 +1,23 @@
 import traceback
 
 import eventlet
-from eventlet import queue
+from eventlet.greenthread import GreenThread
+from eventlet.queue import LightQueue
+from eventlet.semaphore import Semaphore
+from eventlet.event import Event
 from eventlet.support import greenlets as greenlet
 import six
 
 __all__ = ['GreenPool', 'GreenPile']
 
 DEBUG = True
+getcurrent = greenlet.getcurrent
 
 
 class GreenPool(object):
     """The GreenPool class is a pool of green threads.
     """
+    __slots__ = ['size', 'coroutines_running', 'sem', 'no_coros_running']
```

Comment:
This will definitely break a lot of code in other projects. Needs very strong support by numbers.